### DISCUSSION TITLE
Sharing a card

### DISCUSSION THREAD

**Discussion by eigenphase** [rednblkx/HomeKey-ESP32#68] [Q&A] [answered]
I understand this might not be 100% ESP32-HomeKey related but right now it seems to me I can only add the key with one Apple account. This is quite problematic because of other family members and even myself use two accounts.

Likely with a "Hub" it's possible to share with other users. But I neither have nor do I want a Home "Hub" because I run Home Assistant.

Is there any other way to bring the keycard to multiple devices with multiple Apple IDs?

If not, would this be something that could be implemented? Basically having multiple "virtual" HAPs and being able to pair the ESP32-HomeKey multiple times.

**Comment by rednblkx** [comment]
For short, no, however, this should change with iOS 18 as they will introduce “Guest” sharing.

Apple designed the Home Key to be one per Home and was initially meant to be shareable but I guess they abandoned it along the way in favor of the new Aliro protocol that will come with iOS 18.

Currently, there’s no way to share a HomeKey natively and creating multiple “virtual locks” within the same device would create a mess and an problem waiting to happen.

**Reply by eigenphase** [reply]
Wow that's a bummer ... but iOS 18 support would make me very hopeful. Will ESP32-HomeKey be compatible with Aliro is is this something completely different that wouldn't work on iOS18?

> Currently, there’s no way to share a HomeKey natively

However, when creating a "Home Hub" it should work, right?
I have no experience with Apple Home, but at least with a "Home Hub" it should work though, right?
Then my question is:

1. Is there any way to "emulate" a hub with Home Assistant or an extension? Similar to Home Bridge? 
2. If I were to set up a hub with Apple TV or HomePod (I do NOT want) and I would share the lock then, would the virtual keycard appear in the other accounts and can I remote the hub afterwards?

**Reply by rednblkx** [reply, marked as answer]
Two corrections on my previous comment:
1. It’s not clear if Aliro will replace current HomeKey though it is in a way the same thing, however, Home Key will still be Home Key, Aliro will just be used as a backend, possibly only for Matter accessories but we’ll just have to wait for the launch of iOS 18 and see then, I was only alluding to the “Guest” feature which is part of Aliro spec and included in iOS 18
2. We don't currently know how the "Guest" feature will work 

And to answer your questions:

> However, when creating a "Home Hub" it should work, right?
> I have no experience with Apple Home, but at least with a "Home Hub" it should work though, right?

Yes, that's right, ever since they updated the HomeKit Architecture you can no longer add people to your Apple Home without a home hub.

> Is there any way to "emulate" a hub with Home Assistant or an extension? Similar to Home Bridge?

As of now, to my knowledge there is no solution that emulates the functionality of a home hub, though this might be something that i might be working on :wink:  

> If I were to set up a hub with Apple TV or HomePod (I do NOT want) and I would share the lock then, would the virtual keycard appear in the other accounts and can I remote the hub afterwards?

Yes, exactly, after you have added a person to your Apple Home, a key should then shortly appear in their Wallet as you see it yourself.

**Reply by eigenphase** [reply]
> though this might be something that i might be working on 😉

😮🎉🎉
